src/main/resources/public/ts/controllers/controller.ts

```diff
@@ -817,6 +817,16 @@ export const calendarController =  ng.controller('CalendarController',
         };
     }
 
+    $scope.nameOfShareButton = (calendarEvent) : string => {
+        let numberOfSharedCalendars = calendarEvent.calendar
```

Comment:
On est sûr que ton tableau `calendar` sera toujours défini ? pour utiliser le filter sans prendre en compte l'erreur `cannot read property of unefined`

Reply:
Oui, l'évènement appartient forcément à au moins un calendrier qui sera défini dans calendar  (si calendar existe pas l'évènement n'apparaitra pas sur le front)